src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -414,5 +428,76 @@ private static void CheckForCompletedTransfers(SafeCurlMultiHandle multiHandle,
                 }
             }
         }
+
+        private async static void PollFunction(SafeCurlMultiHandle multiHandle)
+        {
+            while (true)
+            {
+                if (multiHandle.CancellationToken.IsCancellationRequested)
+                {
+                    return;
+                }
+
+                var count = multiHandle.FdList.Count;
```

Comment:
Why isn't a lock necessary here when accessing FdList?  If Count happens to be thread-safe, it's an implementation detail of List that we shouldn't count on.